### DISCUSSION TITLE
Numeric citation format: multiple citations within square brackets

### DISCUSSION THREAD

**Discussion by iabraham** [quarto-dev/quarto-cli#5872] [Q&A] [answered]
### Description

My question is exactly [this question](https://tex.stackexchange.com/questions/191418/biblatex-multiple-citation-numbers-all-within-same-bracket-with-the-ieee-style) but with Quarto. In the [example/documentation](https://quarto.org/docs/authoring/footnotes-and-citations.html#sec-citations) we are shown:

```
Blah Blah [@wickham2015; @knuth1984].
```
turn to 
```
Blah Blah [1, 2].
```

However the documentation did not tell which CSL file or YAML option it used to create that particular output. I looked at the Github source for that page and it seems the shown output was hard coded. I am using the [ieee.csl](https://www.zotero.org/styles/ieee) file. This format of clubbing together multiple citations in single brackets is a very common (personally I find `[2,6,9,11]` is more aesthetic than `[2][6][9]]11]` and clearly from the very first link to Stackexchange, many other people do as well). 

How can I do this in Quarto?

**Comment by mcanouil** [comment]
Default CSL in Pandoc/Quarto is "Chicago Manual of Style".

If you are looking for bracketed references, see the links for CSL files in <https://quarto.org/docs/authoring/footnotes-and-citations.html#citation-style>.

**Reply by iabraham** [reply]
Here is a link to the ["Chicago Manual of Style"](https://www.chicagomanualofstyle.org/home.html) just to help myself answer the question.

**Reply by mcanouil** [reply]
I don't understand "just to help myself answer the question." and "😕" ...

There are multiple and various styles which can do the bracket thing.
Were you expecting me to list them for you? In Quarto documentation, you have a link to <https://www.zotero.org/styles> and to <https://github.com/citation-style-language/styles>.
The gallery from Zotero also includes a preview.

For example using "bracket" as keyword:
<img width="1508" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/04bc6d6d-4ce4-4046-9085-2238725f6fc7">

**Reply by iabraham** [reply]
I understand all that; I can follow links and read the documentation. However you have posted a non-answer. 

I specifically asked: _how do you combine two or more citations under single bracket as shown in the documentation_? For example, the IEEE style file  uses the numeric format; **however** if you try using it; it doesn't do what I want:

<img width="1215" alt="Screenshot 2023-06-11 at 2 20 16 PM" src="https://github.com/quarto-dev/quarto-cli/assets/11072297/d1f8e9b1-1d92-4449-a92a-0feaba672536">

"Go read the manual" is perfectly acceptable if completely useless answer to almost any question. If you read the StackExchange link in the original post; you would see that the default IEEE style behaves as above. However, if you provide `numeric-comp` as the citation style it will change its behavior:

```\usepackage[backend=bibtex,bibstyle=ieee,citestyle=numeric-comp]{biblatex}```

How does one do that in Quarto? And once again "Go read the citation language schema and write one `.csl` file yourself" is a perfectly acceptable answer; even if not very useful.

**Reply by mcanouil** [reply]
Let me rephrase, this has nothing to do with Pandoc/Quarto, it's all defined by the CSL file.
Thus to look for CSL doing that and/or modify one.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/b6b63197-58d2-4e22-a2a7-f37c9b737ac9">

**Reply by mcanouil** [reply]
To note, you asked to do "combined bracketed citation" in Quarto, I answered exactly how to do that, i.e., use the syntax from the documentation and a compatible CSL file.

Now you are asking, "how can I have combined bracketed citation using ieee?".
Not at all the same question.

**Comment by iabraham** [comment, marked as answer]
For anyone stumbling across this question; the fix is to slightly modify the `.csl` file (posted here as a .txt file since Github does not recognize the proper extension). 

<img width="1215" alt="Screenshot 2023-06-11 at 2 54 09 PM" src="https://github.com/quarto-dev/quarto-cli/assets/11072297/b33c60f4-8a77-4a38-aa9f-a5cb86547bd4">


[ieee-mod.txt](https://github.com/quarto-dev/quarto-cli/files/11715590/ieee-mod.txt)

**Reply by DbCrWk** [reply]
@iabraham thanks for this solution! It works well for me :)
